### PR TITLE
add support for different chip detection + quad rotary demo

### PR DESCRIPTION
**Pull request by ladyada**
@BlitzCityDIY will need linting n stuffs but is tested on a feather m4

### CODE REVIEW

**Comment by BlitzCityDIY** [comment]
@ladyada thanks! i ran pylint for the example but i wanted to check on an error in **seesaw.py** in the `analog_read` and `analog_write` functions:
```
seesaw.py:268:27: E0602: Undefined variable '_ATTINY8X7_HW_ID_CODE' (undefined-variable)
seesaw.py:364:27: E0602: Undefined variable '_ATTINY8X7_HW_ID_CODE' (undefined-variable)
```

i wasn't sure if the variable was maybe supposed to be `_ATTINY807_HW_ID_CODE` or `_ATTINY817_HW_ID_CODE`?

File: adafruit_seesaw/seesaw.py [hunk 1]
```diff
@@ -164,15 +169,20 @@ def __init__(self, i2c_bus, addr=0x49, drdy=None, reset=True):
             from adafruit_seesaw.robohat import MM1_Pinmap
 
             self.pin_mapping = MM1_Pinmap
-        elif pid in (_5690_PID, _5681_PID, _5743_PID):
+        elif (pid in (_5690_PID, _5681_PID, _5743_PID)) or (self.chip_id in
+                                                           (_ATTINY817_HW_ID_CODE,
+                                                            _ATTINY807_HW_ID_CODE,
+                                                            _ATTINY1617_HW_ID_CODE)):
```
Comment (caternuson) [inline adafruit_seesaw/seesaw.py:175]: Should these be checking for *16 IDs?

**Review by caternuson — CHANGES_REQUESTED** [review]

File: adafruit_seesaw/seesaw.py (hunk 1 above)
Comment (ladyada) [reply adafruit_seesaw/seesaw.py]: fixed

**Comment by ladyada** [comment]
@BlitzCityDIY ok i think if ixed that error, only the samd09 has weird offsets. now we just use the native pin names

File: adafruit_seesaw/seesaw.py [hunk 2]
```diff
@@ -255,10 +265,10 @@ def analog_read(self, pin, delay=0.008):
         if pin not in self.pin_mapping.analog_pins:
             raise ValueError("Invalid ADC pin")
 
-        if self.chip_id == _ATTINY8X7_HW_ID_CODE:
-            offset = pin
-        elif self.chip_id == _SAMD09_HW_ID_CODE:
+        self.chip_id == _SAMD09_HW_ID_CODE:
```
Comment (caternuson) [inline adafruit_seesaw/seesaw.py:268]: Simple syntax error here - missing `if`.

I think that's rippling out and causing a lot of the other failures. Might also need some black/pylint-ification of the two files as well.

**Review by caternuson — CHANGES_REQUESTED** [review]

File: adafruit_seesaw/seesaw.py (hunk 2 above)
Comment (ladyada) [reply adafruit_seesaw/seesaw.py]: *sigh* sorry i thought it pushed, commited & pushed now

**Comment by caternuson** [comment]
@ladyada OK, now it's just black (like clangformat, just run it and let it do what it wants) and pylint (fix or shush, etc).

**Comment by ladyada** [comment]
@BlitzCityDIY can you do the 🧹 - im getting roped into show stuff :)

**Review by caternuson — APPROVED** [review]

**Comment by ladyada** [comment]
thanks all :)

**Pull request merged**